build.gradle.kts

```diff
@@ -198,6 +188,17 @@ kotlin.target.compilations.configureEach {
 }
 
 tasks {
+    withType<KotlinCompilationTask<*>>().configureEach {
```

Comment:
Why did you move to a global withType, rather than configuring the compilation's options/task?

(Compilations are the way forward, we don't need to know about tasks any more; similar to how AGP's variant and artifact DSL hides the existence of tasks.)

Reply:
As noted in the removed comment (in commit 2) -- now that kotlin-dsl is not messing with us -- we can use the cleanest DSL:

```kotlin
kotlin.target.compilations.configureEach {
  compilations.compilerOptions.configure {
    ...
  }
}
```
No need for any task references.

Reply:
> Why did you move to a global withType

Because [that's what the documentation says to do to](https://kotlinlang.org/docs/gradle-compiler-options.html#target-the-jvm) to modify compiler options. Actually it should be `withType<KotlinJvmCompile>()` because we're not in a multiplatform project, I can update.

The actual cleanest config for setting compiler options would be:
```kotlin
kotlin {
    compilerOptions {
        apiVersion = ...
        languageVersion = ...
    }
}
```

But this requires the project to update to Kotlin 1.9.

There's no mention anywhere [here ](https://kotlinlang.org/docs/gradle-configure-project.html) or [here](https://kotlinlang.org/docs/gradle-compiler-options.html) of the syntax that's currently used or what you proposed above?

Reply:
This was the last blocking bit of feedback which I believe I've answered above. Are you happy to approve?